embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/spans/SpanSinkImpl.kt

```diff
@@ -19,16 +21,17 @@ internal class SpanSinkImpl : SpanSink {
     }
 
     override fun completedSpans(): List<EmbraceSpanData> {
-        synchronized(completedSpans) {
-            return completedSpans.toList()
-        }
+        val spansToReturn = completedSpans.size
+        return completedSpans.take(spansToReturn)
```

Comment:
Why is this needed? Instead of just returning `completedSpans.toList()`

Reply:
Because we're not synchronizing this and and write, we can have the completedSpans queue being modified while we are taking a snapshot. Just to be predictable especially for the flush use case, we are doing this so we aren't trying to flush events that were added while the flush was taking place.

Once the flush is triggered, everything that comes after should not be part of what's being flushed.